datafusion/core/src/catalog/mod.rs

```diff
@@ -295,6 +296,118 @@ impl CatalogProvider for MemoryCatalogProvider {
     }
 }
 
+/// Resolve all table references in the SQL statement.
```

Comment:
```suggestion
/// Resolve all table references in the SQL statement.
///
/// Note that this will also return the names of any common 
/// table expressions (CTEs) that are may be defined in the
/// query
```